tests/trainer/test_tp.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import torch
+from packaging import version
+from torch.utils.data import DataLoader
+
+from composer.trainer.trainer import Trainer
+from composer.utils import dist
+from tests.common import (
+    RandomClassificationDataset,
+    SimpleModel,
+    world_size,
+)
+
+
+@pytest.mark.gpu
+@world_size(2)
+@pytest.mark.skip  # TP does not work with DP dimension 1
+@pytest.mark.filterwarnings('ignore:FSDP is switching to use `NO_SHARD`.*:UserWarning')
+@pytest.mark.skipif(version.parse(torch.__version__) < version.parse('2.3'), reason='requires PyTorch 2.3+')
+def test_tp_train(world_size: int):
+    from torch.distributed.tensor.parallel import (  # PrepareModuleInput,; SequenceParallel,; parallelize_module,
+        ColwiseParallel,
+        RowwiseParallel,
+    )
+
+    model = SimpleModel()
+    dataset = RandomClassificationDataset(size=10)
+    dataloader = DataLoader(dataset, sampler=dist.get_sampler(dataset))
+    optimizer = torch.optim.SGD(model.parameters(), lr=0.01)
+
+    layer_plan = {
```

Comment:
Could we test out TP for other layers like norms as well?

Reply:
Any specific reason? it's a bit annoying since we dont have a convenient standard model with norms, but I can add if you think its important

Reply:
For example, we're not testing sequence parallel elementwise ops with TP, which is what we'd want to do in a full TP setup. It would be nice to validate that this is working by adding it to the layer_plan and/or making a model class variant.

Reply:
Discussed offline, because its complicated we're going to push until later when we add to foundry.